charts/warpstream-agent/templates/deployment.yaml

```diff
@@ -151,16 +156,10 @@ spec:
                 fieldRef:
                   apiVersion: v1
                   fieldPath: metadata.namespace
-            {{- if and .Values.resources (not .Values.goMaxProcs) }}
-            {{- if .Values.resources.requests }}
+            {{- if or .Values.resources.requests .Values.goMaxProcs }}
             - name: GOMAXPROCS
               value: {{ $goMaxProcs | quote }}
             {{- end }}
-            {{- end }}
-            {{- if .Values.goMaxProcs }}
-            - name: GOMAXPROCS
-              value: {{ .Values.goMaxProcs | quote }}
-            {{- end }}
           {{- if .Values.extraEnv }}
```

Comment:
The environment variable logic gets simplified by leveraging the conditional logic bypassing the math.